src/smc-webapp/editor-data/generic.tsx

```diff
@@ -67,44 +65,58 @@ interface Props {
   path: string;
 }
 
-class DataGeneric extends Component<Props, {}> {
-  render_hint(): Rendered {
-    const ext = filename_extension(this.props.path);
+function DataGeneric(props: Props) {
```

Comment:
I read a lot about this, and clearly the best way with typescript to make a functional component is:
```
const DataGeneric : React.FC<Props> = (props) => {
```

This correctly checks the types both of the inputs *and* the outputs and makes it very clear that you're defining a React Functional Component (in contract, if I see `function DataGeneric(props: Props) {` I don't even know it is a functional component).  See https://levelup.gitconnected.com/using-react-functional-components-with-hooks-in-typescript-6d61075edddc and also the many components I've converted in the last few weeks.